pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,105 @@
+package node
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configV1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var notReadyTolerationSecondsPath = []string{"apiServerArguments", "default-not-ready-toleration-seconds"}
+var unreachableTolerationSecondsPath = []string{"apiServerArguments", "default-unreachable-toleration-seconds"}
+
+// ObserveNotReadyTolerationSeconds observes the value that should be set for default-not-ready-toleration-seconds
+// api server argument on the basis of provided worker latency profile from config node object.
+func ObserveNotReadyTolerationSeconds(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains default-not-ready-toleration-seconds field.
+		ret = configobserver.Pruned(ret, notReadyTolerationSecondsPath)
+	}()
+
+	nodeLister := genericListers.(configobservation.Listers).NodeLister
+	node, err := nodeLister.Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the node object we are not able to determine worker latency profile
+		return existingConfig, append(errs, err)
+	} else if apierrors.IsNotFound(err) {
+		// if config/v1/node/cluster object is not found, that can be treated as a non-error case
+		return existingConfig, errs
+	}
+
+	// read the observed value
+	var observedDefaultNotReadyTolerationSeconds string
+	switch node.Spec.WorkerLatencyProfile {
+	case configV1.DefaultUpdateDefaultReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.DefaultNotReadyTolerationSeconds)
+	case configV1.MediumUpdateAverageReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.MediumNotReadyTolerationSeconds)
+	case configV1.LowUpdateSlowReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.LowNotReadyTolerationSeconds)
+	// in case of empty worker latency profile, do not update config
+	case "":
+		return existingConfig, errs
+	default:
+		return existingConfig, append(errs, fmt.Errorf("unknown worker latency profile found: %v", node.Spec.WorkerLatencyProfile))
```

Comment:
This would probably be a too extreme case, cause we already added k8s enum validation on the node CRD here:
- https://github.com/openshift/api/blob/441fe135b2fce3b4aac41a1488bea0b79cc6c800/config/v1/0000_10_config-operator_01_node.crd.yaml#L46-L52
- https://github.com/openshift/api/blob/master/config/v1/types_node.go

So essentially k8s API should reject any unknown values though only an empty spec results in `configNodeObj.Spec.workerLatencyProfile=""` and any other updates by operator/user rejects the update (both behaviours were tested manually).

Reply:
During an upgrade, the API server will be installed first, then some other controller could change the current profile based on some data even before this operator would be upgraded to a newer version.

Reply:
as discussed over the chat, it can break backward compatibility. Either document the risk and steps to ensure compatibility in the API or tolerate unknown values here and report an error from the operator.